src/protocol.rs

```diff
@@ -224,6 +239,44 @@ impl<'a> Protocol<'a> {
                 let s = iter.next().ok_or(Error::InvalidProtocolString)?;
                 Ok(Protocol::Memory(s.parse()?))
             }
+            "ip6zone" => {
+                let s = iter.next().ok_or(Error::InvalidProtocolString)?;
+                Ok(Protocol::Ip6zone(Cow::Borrowed(s)))
+            }
+            "ipcidr" => {
+                let s = iter.next().ok_or(Error::InvalidProtocolString)?;
+                Ok(Protocol::Ipcidr(s.parse()?))
+            }
+            "garlic64" => {
+                let s = iter
+                    .next()
+                    .ok_or(Error::InvalidProtocolString)?
+                    .replace('-', "+")
+                    .replace('~', "/");
```

Comment:
why do we need replacing these characters?

Reply:
The garlic protocols are very weird in that there is almost no documentation of them anywhere. But apparently, garlic64 doesn't use classic Base64 encoding, but a similar encoding where `+` is replaced with `-` and `/` with `~`. I don't know why. But to use `multibase::Base::Base64.decode` on the string, we have to have a proper Base64 encoding, so I'm just replacing these characters. You can see in the multiaddr implementations in Go (https://github.com/multiformats/go-multiaddr/blob/414c602b9282d609668bd58c90c800468816d9f4/transcoders.go#L270-L271) and Java (https://github.com/multiformats/java-multiaddr/blob/ffdc5320b4ccb5d5e490b6bce72a8da5dbe775ae/src/main/java/io/ipfs/multiaddr/Protocol.java#L231) that they're doing the same.

Reply:
thanks for taking the time to explain and provide examples!